src/main/java/jp/co/soramitsu/sora/crypto/json/SelectiveDisclosureFactory.java

```diff
@@ -0,0 +1,159 @@
+package jp.co.soramitsu.sora.crypto.json;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import jp.co.soramitsu.sora.crypto.common.Hash;
+import jp.co.soramitsu.sora.crypto.common.HexdigestSaltGenerator;
+import jp.co.soramitsu.sora.crypto.common.SaltGenerator;
+import jp.co.soramitsu.sora.crypto.common.SecurityProvider;
+import jp.co.soramitsu.sora.crypto.merkle.MerkleTree;
+import jp.co.soramitsu.sora.crypto.merkle.MerkleTreeFactory;
+import jp.co.soramitsu.sora.crypto.merkle.MerkleTreeProof;
+import jp.co.soramitsu.sora.crypto.type.DigestTypeEnum;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+
+@RequiredArgsConstructor
+public class SelectiveDisclosureFactory {
+
+  private final ObjectMapper mapper;
+  private final Flattener flattener;
+  private final Saltifier saltifier;
+  private final Hashifier hashifier;
+  private final MerkleTreeFactory merkleTreeFactory;
+
+  public SelectiveDisclosureFactory() {
```

Comment:
not critical, but if possible use Telescope pattern or Builder(with non-null checks on must-have members)